integration/src/test/java/edu/unc/lib/boxc/integration/web/access/SearchEndpointIT.java

```diff
@@ -433,6 +434,112 @@ public void testSearchWithSubjectFilterSelectedDisplaysAvailableLanguages() thro
+            var facetNames = new ArrayList<String>();
+
+            for (JsonNode facet : facetFields) {
+                facetNames.add(facet.get("name").asText());
+            }
+
+            assertSuccessfulResponse(resp);
+            assertEquals(10,facetFields.size());
+            assertEquals(defaultFacets(), facetNames);
+        }
+    }
+
+    @Test
+    public void testSearchWithFacetSelectOnlyShowsThoseFacets() throws Exception {
+        createDefaultObjects();
+        var getMethod = new HttpGet(SEARCH_URL + "/?facetSelect=language%2Csubject&getFacets=true");
+
+        try (var resp = httpClient.execute(getMethod)) {
+            var facetFields = getFacetsFromResponse(resp);
+
+            assertSuccessfulResponse(resp);
+            assertEquals(2,facetFields.size());
+            assertEquals("LANGUAGE", facetFields.get(0).get("name").asText());
+            assertEquals("SUBJECT", facetFields.get(1).get("name").asText());
+        }
+    }
+
+    @Test
+    public void testSearchWithFacetSelectCannotShowDisallowedFacets() throws Exception {
+        createDefaultObjects();
+        var getMethod = new HttpGet(SEARCH_URL + "/?facetSelect=language,subject,content+status&getFacets=true");
+
+        try (var resp = httpClient.execute(getMethod)) {
+            var facetFields = getFacetsFromResponse(resp);
+
+            assertSuccessfulResponse(resp);
+            assertEquals(2,facetFields.size());
+            assertEquals("LANGUAGE", facetFields.get(0).get("name").asText());
+            assertEquals("SUBJECT", facetFields.get(1).get("name").asText());
+        }
+    }
+
+    @Test
+    public void testFacetCountsReflectUserPermissions() throws Exception {
+        createDefaultObjects();
+        createLanguageSubjectObjects();
+        // create admin-restricted collections in different languages
+        collectionFactory.createCollection(adminUnit1,
+                Map.of("title", "Collection Object", "adminGroup", "adminGroup", "languageTerm", "kor"));
+        collectionFactory.createCollection(adminUnit1,
+                Map.of("title", "Collection Object", "adminGroup", "adminGroup", "languageTerm", "por"));
+
+        var getMethod = new HttpGet(SEARCH_URL + "/?facetSelect=language&getFacets=true");
+
+        try (var resp = httpClient.execute(getMethod)) {
+            var facetFields = getFacetsFromResponse(resp);
+            var languageFields = new ArrayList<JsonNode>();
+            facetFields.get(0).get("values").elements().forEachRemaining(languageFields::add);
+
+            assertSuccessfulResponse(resp);
+            assertEquals(2,languageFields.size());
+            // should display only languages associated with public objects: English, Cherokee
+            assertValuePresent(languageFields, 0, "value", "English");
+            assertValuePresent(languageFields, 1, "value", "Cherokee");
+        }
+    }
+
+    private ArrayList<String> defaultFacets() {
```

Comment:
It would probably be appropriate to declare this as a constant at the top of the class. There's a nice built in command for initializing a list with values: `Arrays.asList("PARENT_UNIT", "PARENT_COLLECTION", etc);`

We can also use the SearchFieldKey to get these values, like `SearchFieldKey.PARENT_UNIT.name()`, but I don't feel strongly about it, doesn't add readability and its not like the name can change without changing the enum value.

Reply:
I knew there was a way to do it as a one liner! Thanks. I don't think using the SearchFieldKey would be beneficial here...it seems a little more unclear.